cli/src/pcluster/validators/secret_validators.py

```diff
@@ -13,64 +13,78 @@
 
 from pcluster.aws.aws_api import AWSApi
 from pcluster.aws.common import AWSClientError
-from pcluster.validators.common import FailureLevel, Validator
+from pcluster.validators.common import FailureLevel, SecretArnValidator
 
 
-class MungeKeySecretArnValidator(Validator):
-    """MungeKeySecretArn validator."""
+class MungeKeySecretArnExistsValidator(SecretArnValidator):
```

Comment:
Nitpick: I believe this is generalizable: the validator could have another input parameter with a dictionary of pairs `{ service : secret_resource_type }`. In this case you could reuse this for the other validators. But that's low priority for now. Please put a TODO here to remind ourselves that we can generalize this.

Reply:
Done in: `ArnServiceAndResourceValidator`

Also rewrite `_handle_arn_aws_client_error` function in `common.py`. In the future, if we want to call a API that needs a ARN, we can use this function to avoid most of duplicate codes.

Reply:
See new comment. I believe this is fine for the time being.